src/blocks/GenericEmbed/Component.tsx

```diff
@@ -0,0 +1,183 @@
+'use client'
+
+import getTextColorFromBgColor from '@/utilities/getTextColorFromBgColor'
+import { cn } from '@/utilities/ui'
+import DOMPurify from 'isomorphic-dompurify'
+import { useEffect, useState } from 'react'
+import type { GenericEmbedBlock as GenericEmbedBlockProps } from 'src/payload-types'
+
+type Props = GenericEmbedBlockProps & {
+  className?: string
+  wrapInContainer?: boolean
+}
+
+export const GenericEmbedBlock = ({
+  id,
+  html,
+  backgroundColor,
+  alignContent = 'left',
+  className,
+  wrapInContainer = true,
+}: Props) => {
+  const [sanitizedHtml, setSanitizedHtml] = useState<string | null>(null)
+  const [shouldBeIframe, setShouldBeIFrame] = useState(false)
+  const [iframeHeight, setIframeHeight] = useState<number>(600)
+
+  const bgColorClass = `bg-${backgroundColor}`
+  const textColor = getTextColorFromBgColor(backgroundColor)
+  // const alignClass =
+  //   alignContent === 'center'
+  //     ? 'text-center'
+  //     : alignContent === 'right'
+  //       ? 'text-right'
+  //       : 'text-left'
+
+  useEffect(() => {
+    if (typeof window === 'undefined' || !html) return
+
+    // Normalize problematic quotes that are parsed incorrectly by DOMParser and DOMPurify
+    const normalizedHTML = html.replaceAll('“', '"').replaceAll('”', '"')
+
+    const sanitized = DOMPurify.sanitize(normalizedHTML, {
```

Comment:
Does isomorphic-dompurify offer any erroring for bad scripts or ill formatted code? It might be nice to display an error if something isnt right but if it's not easy to tell, probably not worth the time